src/prompts/abstract.ts

```diff
@@ -15,20 +17,37 @@ type Stdout = NodeJS.WriteStream & {
   fd: 1;
 }
 
-export class AbstractPrompt<T> {
+export interface AbstractPromptOptions {
+  stdin?: Stdin;
+  stdout?: Stdout;
+  message: string;
+  timeout?: number;
```

Comment:
Why not using an AbortSignal instead of number?

Reply:
I planned to have both options `signal` & `timeout`.
Do you think we should only keep signal ?

Reply:
Since we can do `AbortSignal.timeout(1000)` I think yes.